DDD-12.md

```diff
@@ -0,0 +1,322 @@
+
+### Quarkus Debezium Lifecycle Events
+
+We can summarize the lifecycle of a Debezium Embedded Engine in the follows steps:
+
+| Phase            | description                                                           | code                                  |
+|------------------|-----------------------------------------------------------------------|---------------------------------------|
+| *initialization* | configuration is built and the engine is created, but not yet running | after the configuration is validated  |
+| *startup*        | connectors are initialized, DB connection established                 | right after `engine.run()` is invoked |
+| *shutdown*       | engine terminates (graceful or with error)                            | observable via `CompletitionCallback` |
+
+The Quarkus Debezium extension allows you to be notified of the engine's state using the following annotation:
+
+```java
+import io.debezium.engine.source;
+import jakarta.enterprise.context.ApplicationScoped;
+
+
+@ApplicationScoped
+class DebeziumEngineLifeCycle {
+    
+    @DebeziumEngineInit()
+    public void init() {
+        /// some logic to apply
+    }
+
+    @DebeziumEngineStartup()
+    public void startup(DebeziumSourceConnectorContext context) {
+        /// some logic to apply 
+    }
+
+    @DebeziumEngineShutdown()
+    public void shutdown(Status status) {
+        /// some logic to apply
+    }
+}
+```
+
+### Quarkus Debezium Heartbeats events
+
+In Debezium, heartbeat events are lightweight, periodic messages emitted when no database changes occur. They confirm that the connector is alive and still connected to the source, helping to detect liveness. In the Quarkus Debezium Extension can detect heartbeats events in this way:
+
+```java
+import io.debezium.engine.ChangeEvent;
+import jakarta.enterprise.context.ApplicationScoped;  
+
+
+@ApplicationScoped  
+class HeartbeatListener {
+  
+    @DebeziumHeartbeat()  
+    public void heartbeat(ChangeEvent<String, String> event) {  
+        /// some logic to apply 
+    }  
+}
+```
+
+it's possible to define the `heartbeat.action.query` and `heartbeat.interval.ms` interval in case of relational databases:
+
+```java
+import io.debezium.engine.ChangeEvent;
+import jakarta.enterprise.context.ApplicationScoped;  
+
+
+@ApplicationScoped  
+class HeartbeatListener {
+  
+    @DebeziumHeartbeat(query="SELECT now()", interval=10000)  
+    public void heartbeat(ChangeEvent<String, String> event) {  
+        /// some logic to apply 
+    }  
+}
+```
+
+### Quarkus Debezium Listener
+
+a Quarkus Developer using a `Debezium Listener`  can intercept events (`INSERT, UPDATE, DELETE...`) from a table like `order`, with a simple annotation like:
+
+```java
+import io.debezium.engine.ChangeEvent;
+import jakarta.enterprise.context.ApplicationScoped;  
+
+
+@ApplicationScoped  
+class OrderListener {
+  
+    @DebeziumListener("order")  
+    public void listener(ChangeEvent<String, String> event) {  
```

Comment:
Are we going also support exposition of raw `SourceRecord` object?

Reply:
Probably we should support it. WDYT?

Reply:
I'd say so, it is one of the things that could be useful for Debezium Server.